src/Microsoft.Framework.FileSystemGlobbing/Internal/PatternContexts/PatternContextRaggedExclude.cs

```diff
@@ -13,19 +13,18 @@ public PatternContextRaggedExclude(IRaggedPattern pattern)
         {
         }
 
-        public override bool Test(FileInfoBase file)
+        public override PatternTestResult Test(FileInfoBase file)
```

Comment:
Is there a good way to have just one copy of this method - the one PatternContextRaggedInclude.cs seems identical. Also do you mind improving the exception message?